iree/tools/CMakeLists.txt

```diff
@@ -6,6 +6,11 @@
 
 # Doesn't use bazel_to_cmake because of various special logic throughout.
 
+# Tools has thread dependency from iree::hal::drivers and absl
+if (NOT ${IREE_ENABLE_THREADING})
```

Comment:
I think we should be able to support at least some of these tools without threading, at least on build targets with file systems / fully featured operating systems. Possibly outside the scope of this change.

```suggestion
# Tools has thread dependency from iree::hal::drivers and absl
# TODO(#3848): drop absl dep
if(NOT ${IREE_ENABLE_THREADING})
```

Reply:
iree:hal::drivers still has thread dependency afterwards, it is the similar situation as `iree/runtime`. Fix #3848 will not remove this condition unfortunately.